src/components/filter/FilterOptions.tsx

```diff
@@ -123,6 +122,14 @@ function FilterOptions({ onFilteredSources }: FilterOptionsProps) {
       if (shouldDelete) {
         tempSet.delete(source._id.toString());
       }
+
+      if (onlyShowActiveSources && source.status === 'gone') {
+        shouldDelete = true;
+      }
+
+      if (shouldDelete) {
+        tempSet.delete(source._id.toString());
+      }
```

Comment:
This looks identical rows 118-124, so probably there's been som accidental duplication